File: stripe/src/main/java/com/stripe/android/model/CvcTokenParams.java
package com.stripe.android.model;

import android.support.annotation.NonNull;

import java.util.HashMap;
import java.util.Map;

public final class CvcTokenParams implements StripeParamsModel {
    private final String mCvc;

    public CvcTokenParams(String cvc) {
        this.mCvc = cvc;
    }

    @NonNull
    @Override
    public Map<String, Object> toParamMap() {
        final Map<String, Object> tokenParams = new HashMap<>();
        tokenParams.put("cvc", mCvc);
        final Map<String, Object> cvcParams = new HashMap<>();
        cvcParams.put(Token.TokenType.CVC_UPDATE, tokenParams);
        return cvcParams;
    }
}

File: stripe/src/main/java/com/stripe/android/model/CvcTokenParams.kt
package com.stripe.android.model

import kotlinx.parcelize.Parcelize

@Parcelize
data class CvcTokenParams(
    private val cvc: String
) : TokenParams(Token.Type.CvcUpdate) {
    override val typeDataParams: Map<String, Any>
        get() = mapOf("cvc" to cvc)
}
